src/libratbag-hidraw.h

```diff
@@ -183,6 +183,33 @@ int ratbag_hidraw_read_input_report(struct ratbag_device *device, uint8_t *buf,
  */
 int ratbag_hidraw_read_input_report_index(struct ratbag_device *device, uint8_t *buf, size_t len, int hidrawno);
 
+/**
+ * Read an input report from the device util a specified filter function is successfull
+ *
+ * @param device the ratbag device
+ * @param[out] buf resulting raw data
+ * @param len length of buf
+ * @param filter filter function
+ *
+ * @return count of data transfered, or a negative errno on error
+ */
+int ratbag_hidraw_read_input_report_filter(
+	struct ratbag_device *device, uint8_t *buf, size_t len, int *(*filter)(uint8_t *buf, size_t len));
```

Comment:
/me stares hard at `int *(*filter)` and wonders if this returns an `int*`?

one day I'll manage to read C function pointer declarations without having to think about it.